src/m_misc2.c

```diff
@@ -167,6 +167,14 @@ char *M_DirName(const char *path)
     pf = strrchr(path, '/');
 #ifdef _WIN32
     pb = strrchr(path, '\\');
+    if (pf == NULL && pb == NULL)
```

Comment:
Does this work for you? Problem is that `M_DirName()` "returns the directory portion of the given path, without the trailing slash separator character". This means that the function returns `C:` for both `C:\Doom` and `C:Doom` arguments.

Reply:
I'm not sure how to test this. We use `M_DirName` function only in one place: https://github.com/fabiangreffrath/woof/blob/8d4ccf9a523559e552f6713ff521defa8ebf6a0e/src/d_iwad.c#L540

Reply:
And in this PR also in `D_DoomExeDir()`.

Reply:
Anyway, it works fine for me so far.